controllers/clusterrelocation_controller.go

```diff
@@ -328,12 +337,70 @@ func (r *ClusterRelocationReconciler) finalizeRelocation(ctx context.Context, lo
 		if err := reconcileAPI.Cleanup(ctx, r.Client, logger); err != nil {
 			return err
 		}
+
+		clusterDNS := &configv1.DNS{}
+		if err := r.Client.Get(ctx, types.NamespacedName{Name: "cluster"}, clusterDNS); err != nil {
+			return err
+		}
+		if err := r.verifyDomain(ctx, clusterDNS.Spec.BaseDomain, logger); err != nil {
+			return err
+		}
 	}
 
 	logger.Info("Successfully finalized ClusterRelocation")
 	return nil
 }
 
+func (r *ClusterRelocationReconciler) verifyDomain(ctx context.Context, domainName string, logger logr.Logger) error {
+	urls := []map[string]string{
+		{
+			"type":       "ingress",
+			"url":        fmt.Sprintf("test.apps.%s:443", domainName),
+			"commonName": fmt.Sprintf("*.apps.%s", domainName),
+		},
+		{
+			"type":       "kube-apiserver",
+			"url":        fmt.Sprintf("api.%s:6443", domainName),
+			"commonName": fmt.Sprintf("api.%s", domainName),
+		},
+	}
+
+	for _, v := range urls {
+		updated := false
+		for {
+			conn, err := tls.Dial("tcp", v["url"], &tls.Config{InsecureSkipVerify: true})
+			if err != nil {
+				return err
+			}
+			certs := conn.ConnectionState().PeerCertificates
+			conn.Close()
+			for _, cert := range certs {
+				if cert.Subject.CommonName == v["commonName"] {
+					updated = true
+				}
+			}
+			if updated {
+				// ensure that ClusterOperator has settled
+				if err := util.WaitForCO(ctx, r.Client, logger, v["type"]); err != nil {
+					return err
+				}
+				break
+			} else {
+				logger.Info(fmt.Sprintf("Waiting for %s to update", v["type"]))
+				time.Sleep(time.Second * 10)
+			}
+		}
+	}
+
+	if err := util.WaitForCO(ctx, r.Client, logger, "openshift-apiserver"); err != nil {
+		return err
+	}
+	if err := reconcileIngress.ResetRoutes(ctx, r.Client, fmt.Sprintf("apps.%s", domainName), logger); err != nil {
+		return err
+	}
```

Comment:
It's still strange to reset (delete and expect them to get recreated) routes in a function called VerifyRoutes, consider moving it out or renaming the function.